apps/apps.c

```diff
@@ -2340,6 +2340,72 @@ int app_isdir(const char *name)
 }
 #endif
 
+/* app_dirname section */
+
+/*
+ * This exactly follows what POSIX's
+ * dirname does, but is implemented
+ * in a more platform independent way.
+ *
+ * path        dirname
+ * /usr/lib    /usr
+ * /usr/       /
+ * usr         .
+ * /           /
+ * .           .
+ * ..          .
+ * ""          .
+ *
+ * Note: this function also keeps the
+ * possibility of modifying the 'path'
+ * string same as POSIX dirname.
+ */
+static char *posix_dirname(char *path)
```

Comment:
Should this actually return "const char *"?

Reply:
It could be. But the `app_dirname` was defined following the `dirname(3)` which returns `char *`, so both `app_dirname` and `posix_dirname` are defined without `const`.

Reply:
Ok, fair enough.